### ISSUE TITLE
Why removing p- prefix?

### ISSUE THREAD

**Issue by FadiNouh1** [primefaces/primeflex#69]
I think it was wrong idea about removing the p- prefix from the classes of the primeflex.
This require big change in many projects which are using the library

**Comment by melloware** [comment]
@FadiNouh1 We have written a migration tool you can use which will convert your whole project.  

See: https://github.com/primefaces/primefaces/tree/master/primefaces-cli

**Comment by elielcena** [comment]
@melloware 
When are you going to update the primeng showcase?
There it is still with "p-", this can generate confusion, because the user installs the dependency in the project, will use the example and it doesn't work.

**Comment by melloware** [comment]
@elielcena I don't work for Primetek I am just an open source committer on the PrimeFaces JSF project.

**Comment by AmazingDreams** [comment]
All themes are still using p-grid too https://github.com/primefaces/primeng/search?l=CSS&q=p-grid

**Comment by bisubus** [comment]
The problem with the lack of a namespace is that... there is no namespace. SASS doesn't support importing classes without leaking them to global scope, and the libary doesn't provide SASS placeholders that would allow to use custom class names. This makes the library useless for the use in scopes you don't own - a module in your company or a widget on somebody else's website. It would be a good idea to prefix all class names with optional variable.

**Comment by Schmaga** [comment]
This could indeed be a major problem. 

I know that many customers are using bootstrap for its grid system, but primeng for some of its components. I'm not sure, is there a fixed dependency from primeng to primeflex for some layouting? But if there is, this will totally block the upgrade to the new primeflex, because its class names would now clash with many bootstrap class names.

**Comment by GeertHuls** [comment]
> All themes are still using p-grid too https://github.com/primefaces/primeng/search?l=CSS&q=p-grid
> 
> I think it is a really bad idea
> 
> At this point it is not possible to upgrade, as PrimeNG is not compatible with PrimeFlex
> 
> [primefaces/primeng#10469](https://github.com/primefaces/primeng/issues/10469)

I'm using PrimeFlex 3.0 within my PrimeNG 12 project without any issues. Prime flex is just a css utility library which can be used with any kind of front-end framework. The only thing you'll have to do is to figure out how to convert to old naming conventions to the new ones which isn't difficult at all with help of its excellent [docs](https://www.primefaces.org/primeflex).

**Comment by melloware** [comment]
We have a PrimeFaces CLI that will convert your entire project for you if you look at the PrimeFaces repo. The Readme shows you how to use it.

**Comment by andrewlorenz** [comment]
I agree with @bisubus .  It is a seriously unfounded thing to say (as @cagataycivici did above) that "_we are removing p- because look how silly it gets when you use lots of classes in combination_":
```
<div class="p-bg-blue-500 p-text-white p-p-3 p-m-2 p-sm-w-6 p-lg-col-12 p-absolute p-top-0">
```
So what?  These are utility classes, **exactly** the reason why there may be a lot, and at least they can (could) be isolated from the global styling namespace.  I personally use 3 character prefixes for my project classes, for exactly the same reason.
I'd rather have my own issues with readability - and how long my lines are in my editor - these are not a package's concern!

**Comment by Jagan-P** [comment]
> > All themes are still using p-grid too https://github.com/primefaces/primeng/search?l=CSS&q=p-grid
> > I think it is a really bad idea
> > At this point it is not possible to upgrade, as PrimeNG is not compatible with PrimeFlex
> > [primefaces/primeng#10469](https://github.com/primefaces/primeng/issues/10469)
> 
> I'm using PrimeFlex 3.0 within my PrimeNG 12 project without any issues. Prime flex is just a css utility library which can be used with any kind of front-end framework. The only thing you'll have to do is to figure out how to convert the old naming conventions to the new ones which isn't difficult at all with help of its excellent [docs](https://www.primefaces.org/primeflex).

I guess you haven't used many of the PrimeNG components. Internally PrimeNG components (the template html of components) still use old classes. So you cannot remove primeflex 2 from your project completely. 

You can see a demo here what it causes [https://github.com/primefaces/primeng/issues/10469](url)

Since this is just css it can be fixed by having both Primeflex v2 and primeflex v3 in your project. This doesn't cause any issues as long as you are using any other css-framework like bootstrap.

**Comment by GeertHuls** [comment]
> > > All themes are still using p-grid too https://github.com/primefaces/primeng/search?l=CSS&q=p-grid
> > > I think it is a really bad idea
> > > At this point it is not possible to upgrade, as PrimeNG is not compatible with PrimeFlex
> > > [primefaces/primeng#10469](https://github.com/primefaces/primeng/issues/10469)
> > 
> > 
> > I'm using PrimeFlex 3.0 within my PrimeNG 12 project without any issues. Prime flex is just a css utility library which can be used with any kind of front-end framework. The only thing you'll have to do is to figure out how to convert the old naming conventions to the new ones which isn't difficult at all with help of its excellent [docs](https://www.primefaces.org/primeflex).
> 
> I guess you haven't used many of the PrimeNG components. Internally PrimeNG components (the template html of components) still use old classes. So you cannot remove primeflex 2 from your project completely.
> 
> You can see a demo here what it causes [https://github.com/primefaces/primeng/issues/10469](url)
> 
> Since this is just css it can be fixed by having both Primeflex v2 and primeflex v3 in your project. This doesn't cause any issues as long as you are using any other css-framework like bootstrap.

Actually I am building an app with PrimeNG at the moment, so far I haven't encountered any issues yet. Then again, at the moment I've only got a handful of components so that might be the reason. :)

Looking at template code in your stackblitz example:
https://stackblitz.com/edit/primeng-dataview-demo-f6fbie?file=src%2Fapp%2Fapp.component.html

TBH I don't see the issue there either. The only thing is noticed is the **styleClass="p-mb-2 p-mb-md-0"** on the dropdown in the header template on line 7. But that code is under your control as a content projection template so you could easily change it to **styleClass="mb-2 md:mb-0"**, I guess.

Perhaps I'm missing something. Can you give an example of old classes being used in the p-dataview component?

**Comment by Jagan-P** [comment]
> 

Nope. I mean the html of primeng (of its own). The one's that will be in the node_modules of your project. They agreed that they are aware of the issue and is on their roadmap 5 months ago. Still no update

**Comment by Jagan-P** [comment]
> > > > All themes are still using p-grid too https://github.com/primefaces/primeng/search?l=CSS&q=p-grid
> > > > I think it is a really bad idea
> > > > At this point it is not possible to upgrade, as PrimeNG is not compatible with PrimeFlex
> > > > [primefaces/primeng#10469](https://github.com/primefaces/primeng/issues/10469)
> > > 
> > > 
> > > I'm using PrimeFlex 3.0 within my PrimeNG 12 project without any issues. Prime flex is just a css utility library which can be used with any kind of front-end framework. The only thing you'll have to do is to figure out how to convert the old naming conventions to the new ones which isn't difficult at all with help of its excellent [docs](https://www.primefaces.org/primeflex).
> > 
> > 
> > I guess you haven't used many of the PrimeNG components. Internally PrimeNG components (the template html of components) still use old classes. So you cannot remove primeflex 2 from your project completely.
> > You can see a demo here what it causes [https://github.com/primefaces/primeng/issues/10469](url)
> > Since this is just css it can be fixed by having both Primeflex v2 and primeflex v3 in your project. This doesn't cause any issues as long as you are using any other css-framework like bootstrap.
> 
> Actually I am building an app with PrimeNG at the moment, so far I haven't encountered any issues yet. Then again, at the moment I've only got a handful of components so that might be the reason. :)
> 
> Looking at template code in your stackblitz example: https://stackblitz.com/edit/primeng-dataview-demo-f6fbie?file=src%2Fapp%2Fapp.component.html
> 
> TBH I don't see the issue there either. The only thing is noticed is the **styleClass="p-mb-2 p-mb-md-0"** on the dropdown in the header template on line 7. But that code is under your control as a content projection template so you could easily change it to **styleClass="mb-2 md:mb-0"**, I guess.
> 
> Perhaps I'm missing something. Can you give an example of old classes being used in the p-dataview component?

Upon going through the source code I observed that primeng is made compatible with primeflex v2 and v3 from the major release v12.

So the issue I reported 5 months ago is fixed [10469](https://github.com/primefaces/primeng/issues/10469) 

Not sure about themes though which @AmazingDreams has mentioned.

**Comment by siriussoftware** [comment]
If this is such a trivial change and has no great, why is the showcase and examples still using the p prefix.

**Comment by swotiendang** [comment]
I overcame this issue by adding ```$prefix: 'p-' !default; // override primeflex variable``` before @import primeflex.scss
